src/controllers/clientController.js

```diff
@@ -0,0 +1,124 @@
+const Client = require('../models/client');
+const Dish = require('../models/dish');
+
+exports.createClient = async (req, res) => {
+  try {
+    const { user, address, contactNumber } = req.body;
+
+    const client = new Client({
+      user,
+      address,
+      contactNumber,
+    });
+
+    const createdClient = await client.save();
+
+    res.status(201).json(createdClient);
+  } catch (error) {
+    res.status(500).json({ error: 'Server error' });
+  }
+};
+
+exports.getClient = async (req, res) => {
+  try {
+    const clientId = req.body.id;
+    const client = await Client.findById(clientId);
+
+    if (!client) {
+      res.status(404).json({ error: 'Client not found' });
+    }
+
+    res.status(200).json(client);
+  } catch (error) {
+    res.status(500).json({ error: 'Server error' });
+  }
+};
+
+exports.updateClient = async (req, res) => {
+  try {
+    const clientId = req.body.id;
+    const client = await Client.findById(clientId);
+
+    if (!client) {
+      res.status(404).json({ error: 'Client not found' });
+    }
+
+    client.address = req.body.address || client.address;
+    client.contactNumber = req.body.contactNumber || client.contactNumber;
+
+    if (req.body.favoriteDishes && Array.isArray(req.body.favoriteDishes)) {
+      client.favoriteDishes = req.body.favoriteDishes;
+    }
```

Comment:
Client shouldn't be able to update the favorite dish right? because the favorite dish is something you need to add or delete from the list and that is it.

Reply:
sorry I thought updating includes favorite dish as well, we can change it no problem